### ISSUE TITLE
Integration stops getting data for a week at a time

### ISSUE THREAD

**Issue by dziban303** [hokiebrian/eia_hourly_demand#3]
I've been using this integration for about a year and recently it's been kind of flaky. I'm getting data for MISO; so far this month there was an outage from 7 June through 16 June and from the 22nd through to the present, as well as a few single days here and there. The data is there in the API. I checked the logs and this error is present:

```
homeassistant  | 2024-06-24 09:19:05.870 ERROR (MainThread) [homeassistant.components.sensor] eia_hourly_demand: Error on device update!
homeassistant  | Traceback (most recent call last):
homeassistant  |   File "/config/custom_components/eia_hourly_demand/sensor.py", line 65, in async_update
homeassistant  |     async with session.get(url, timeout=timeout) as response:
homeassistant  |   File "/usr/local/lib/python3.12/site-packages/aiohttp/client.py", line 1197, in __aenter__
homeassistant  |     self._resp = await self._coro
homeassistant  |                  ^^^^^^^^^^^^^^^^
homeassistant  |   File "/usr/local/lib/python3.12/site-packages/aiohttp/client.py", line 507, in _request
homeassistant  |     with timer:
homeassistant  |   File "/usr/local/lib/python3.12/site-packages/aiohttp/helpers.py", line 735, in __exit__
homeassistant  |     raise asyncio.TimeoutError from None
homeassistant  | TimeoutError
homeassistant  |
homeassistant  | During handling of the above exception, another exception occurred:
homeassistant  |
homeassistant  | Traceback (most recent call last):
homeassistant  |   File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 729, in _async_add_entity
homeassistant  |     await entity.async_device_update(warning=False)
homeassistant  |   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1289, in async_device_update
homeassistant  |     await self.async_update()
homeassistant  |   File "/config/custom_components/eia_hourly_demand/sensor.py", line 79, in async_update
homeassistant  |     except asyncio.TimeoutError as e:
homeassistant  |            ^^^^^^^
homeassistant  | NameError: name 'asyncio' is not defined. Did you forget to import 'asyncio'
```

Any ideas? I'm using Home Assistant 2024.6.4 in Docker.

**Comment by hokiebrian** [comment]
I just pushed out a small update. The error you're getting should be resolved. I think I had some lingering references to asyncio that are no longer valid. As far as the missing data, I experience the same with one of the instances I follow (DUK), but not any of the others... I've spent some, but not much, time looking at it but haven't come up with anything yet. I'll keep digging. Thank you for raising the issue!
